### ISSUE TITLE
[UX] Move new password show/hide toggle to inside the password fields.

### ISSUE THREAD

**Issue by serundeputy** [backdrop/backdrop-issues#3046]
## Describe your issue or idea

I was on the internet filling out a account sign up form and they had there password view/hide toggle in the password field and I liked it.

### Steps to reproduce (if reporting a bug)

Either install a Backdrop site or visit `/user/%/edit` page.

### Actual behavior (if reporting a bug)

Password toggle is a link above the password field.

### Expected behavior (if reporting a bug)

Move the button inside the password field.
<img width="1644" alt="screen shot 2018-04-21 at 9 52 21 am" src="https://user-images.githubusercontent.com/6763671/39084905-195baf44-454a-11e8-94b6-ba118ef542b2.png">

<img width="1644" alt="screen shot 2018-04-21 at 9 53 15 am" src="https://user-images.githubusercontent.com/6763671/39084909-1fefb6ca-454a-11e8-861a-0d92d6585d72.png">

### Relevant version/system information (if applicable)

`1.x`

**Comment by BWPanda** [comment]
I like this, looks very nice! However I don't recall seeing *any* toggle link in Backdrop... Nothing in FF on Android when I checked just now, and I don't recall seeing anything on the desktop previously. So I think this task is more about adding a link, rather than moving one. But I'm all for it.

**Comment by serundeputy** [comment]
@BWPanda the toggle feature is in `1.x` and will ship with `1.10.0`. https://github.com/backdrop/backdrop-issues/issues/297

**Comment by BWPanda** [comment]
@serundeputy Ah! Thanks for the clarification.

**Comment by quicksketch** [comment]
@serundeputy I agree this looks better, but there are two things that give me pause:

- Password managers like LastPass inject themselves into password fields for their autofill option. Such as this (this is Ars Technica): 
![image](https://user-images.githubusercontent.com/100206/39110545-670de2e2-4686-11e8-9499-084f6afc2b02.png)
- And Edge browser actually provides a password toggle already built into the browser (the only browser I know of that does so):
![image](https://user-images.githubusercontent.com/100206/39110652-ce9968e6-4686-11e8-9a2c-bbf38196b34a.png)

So potentially we have the problem of overlapping and bumping into these common features. Personally, I turn off the LastPass option (but it is on by default). In Edge, we probably should attempt feature-detection and ignore the `#password_toggle` option if it's provided by the web browser directly.

**Comment by jenlampton** [comment]
Another thing we considered initially when putting the toggle outside (and before) the password field was how screen-readers would interact with the element. 

We wanted non-visual people to know that there was a password toggle option before they started typing in the field, so having it come first in the markup is important. We may still be able to do this and have it display visually  on the right side of the inside of password field -- if we decide we want to -- but it is something that will need to be tested thoroughly.

I personally prefer the tool the way we have it now for the reasons @quicksketch mentioned above.
